rosplan_knowledge_base/src/VALVisitorProblem.cpp

```diff
@@ -80,25 +109,22 @@ namespace KCL_rosplan {
                      varSymbolIterator != tempPredicate->getArgs()->end(); varSymbolIterator++) {
 
                     const VAL::var_symbol * tempVarSymbol = *varSymbolIterator;
-
                     for ( unsigned int a = 0; a < last_prop.typed_parameters.size(); a++){
 
                         // compare predicate symbol name to match the value of the typed_parameter
-                        if (tempVarSymbol->type->getName() == last_prop.typed_parameters[a].value) {
+                        if (tempVarSymbol->type->getName() == last_prop.typed_parameters[a].value && tempVarSymbol->pddl_typed_symbol::getName() == last_prop.labels[a]) {
 
                             diagnostic_msgs::KeyValue param;
-                            param.key = tempVarSymbol->pddl_typed_symbol::getName();
+                            param.key = last_prop.labels[a];
```

Comment:
As above, I need to check if this is the type of the label. If it is the type, then the KnowledgeComparitor (which currently checks both label and value) will not match any facts read from the problem parser, as it is checking for Key=Label, Value=Object, e.g.:
KnowledgeItem
  attribute_name = "robot_at"
  params
    "r"->"kenny"